fix: load config on macOS

bndw opened the pull request:
On macOS, `os.UserConfigDir` returns `$HOME/Library/Application Support` and not `~/.config`. This patch uses a module that works across OSes.

Fixes #3

mattn commented:
If you have problems on macOS, please apply the code only on macOS. This change will break Windows.

bndw commented:
Good catch. I've moved the macOS-specific change to only target macOS, and made the original code not target macOS.

Let me know if you prefer a different implementation approach. I don't love how this splits the Config stuff into multiple files but I wasn't sure of a better way to do it. I can also refactor `loadConfig` a bit so there's not as much duplication.

junkpiano commented on load_config.go:
```diff
@@ -0,0 +1,50 @@
+)
+
+func loadConfig(profile string) (*Config, error) {
+	dir, err := os.UserConfigDir()
```
Hi there,

Allow me to jump in.
Does it also make sense to create a function to return the config directory?

```
func configDir() (string, error) (
    if runtime.GOOS == "darwin" {
        return os.UserHomeDir()
    }

    return os.UserConfigDir()
}
```

Then we don't need separate files per platform.

bndw replied:
Great call, updated.

mattn replied:
homedir is not required at least?

bndw replied:
`homedir` or similar is required for macOS - the example code above is copypasta and doesn't actually work as is.

Again, `os.UserHomeDir()` returns `$HOME/Library/Application Support` and not `~/` on macOS.

junkpiano replied:
Hmm, it's strange. On my mac machine, `os.UserHomeDir()` returns the same value as `$HOME`.
I am on go `1.20`.

bndw replied:
Sorry, I misread `os.UserHomeDir` for `os.UserConfigDir`. The latter is what algia uses today and causes the problem, but you're totally correct that `os.UserHomeDir` returns the homedir on macOS.

Updated the PR again to only use package `os`.

bndw committed:
fix: load config on macOS

On macOS, `os.UserConfigDir` returns `$HOME/Library/Application Support`
and not `~/.config`.

Fixes #3

mattn commented:
thank you
